audio/param.rs

```diff
@@ -408,6 +415,25 @@ impl AutomationEvent {
                 *value = val + (event_start_value - val) * exp.exp() as f32;
                 true
             }
+            AutomationEvent::SetValueCurveAtTime(ref values, start, duration) => {
+                let time_diff = ((duration.0 as f32) - (start.0 as f32)) as f32;
+                let mut progress = ((((current_tick.0 as f32) - (start.0 as f32)) as f32) / time_diff) as f32;
+                if progress < 0.0 {
+                    progress = 0.0 as f32;
+                }
+                let n = values.len() as f32;
+	            let k = (((n - 1.) * progress) as f32).floor();
+                let next = k + 1. as f32;
+                let step = time_diff / (n - 1.);
+                if next < n {
+                    let time_k = (k * step) as f32;
```

Comment:
This is recalculating stuff we already know, an easier way to do this is to calculate `k_float` with `k = k_float.floor()`, and then interpolate between `v[k]` and `v[k+1] with progress `k_float - k`.